client/js/widgets/scoreboard.js

```diff
@@ -48,6 +48,54 @@ class Scoreboard extends Widget {
     return p;
   }
 
+  async click(mode='respect') {
+    if(!await super.click(mode)) {
+      const scoreProperty = this.get('scoreProperty');
+      const seats = this.getIncludedSeats();
+      let players = Object.keys(seats).map(function(s) { return { text: s, value: s }; });
+      if(Array.isArray(seats))
+        players = seats.map(function(s) { return { value: s.get('id'), text: s.get('player') }; });
+
+      const rounds = this.getRounds(seats, scoreProperty, 1).map(function(r, i) { return { text: r, value: i+1 }; });
+
+      if(!players.length || !rounds.length)
+        return;
+
+      try {
+        const result = await this.showInputOverlay({
+          header: 'Set score',
+          fields: [
+            {
+              type: 'select',
+              label: Array.isArray(seats) ? 'Player' : 'Team',
+              options: players,
+              variable: 'player'
+            },
+            {
+              type: 'select',
+              label: this.get('roundLabel'),
+              options: rounds,
+              variable: 'round'
+            },
+            {
+              type: 'number',
+              label: 'Score',
+              variable: 'score'
+            }
+          ]
+        });
+        const seat = Array.isArray(seats) ? widgets.get(result.player) : seats[result.player][0];
+        let scores = seat.get(scoreProperty);
+        if(Array.isArray(scores))
+          scores = [...scores];
+        else
+          scores = [];
+        scores[result.round-1] = result.score;
+        await seat.set(scoreProperty, scores);
```

Comment:
Here, I don't think you should be converting a non-array score in a seat to an array - just change the amount to the given score.